consensus/istanbul/backend/backend.go

```diff
@@ -273,18 +273,20 @@ type Backend struct {
 	vph *validatorPeerHandler
 }
 
+// IsProxy returns if instance has proxy flag
 func (sb *Backend) IsProxy() bool {
-	return sb.proxiedPeer != nil
+	return sb.config.Proxy
 }
 
+// IsProxiedValidator returns if instance has proxied validator flag
 func (sb *Backend) IsProxiedValidator() bool {
-	return sb.proxyNode != nil && sb.proxyNode.peer != nil
+	return sb.config.Proxied
 }
 
 // SendDelegateSignMsgToProxy sends an istanbulDelegateSign message to a proxy
 // if one exists
 func (sb *Backend) SendDelegateSignMsgToProxy(msg []byte) error {
-	if !sb.IsProxiedValidator() {
+	if !sb.IsProxiedValidator() || sb.proxyNode.peer == nil {
```

Comment:
panic if `sb.ProxyNode == nil` here ?

Reply:
I could see this state happening if a validator & proxy disconnection happens right after a validator receives a delegate sign message from the proxy, so I don't think we should purposely cause a validator to crash

Reply:
what I mean is it would panic if `sb.ProxyNode == nil`

Reply:
consider ```if !sb.IsProxiedValidator() || sb.proxyNode == nil || sb.proxyNode.peer == nil {```

Reply:
Ahh good catch